Buildings/Controls/OBC/ASHRAE/PrimarySystem/BoilerPlant/Staging/SetPoints/SetpointController.mo

```diff
@@ -0,0 +1,596 @@
+within Buildings.Controls.OBC.ASHRAE.PrimarySystem.BoilerPlant.Staging.SetPoints;
+block SetpointController
+  "Calculates the boiler stage status setpoint signal"
+
+  parameter Boolean primaryOnly = false
+    "Is the boiler plant a primary-only, condensing boiler plant?"
+    annotation(Dialog(tab="General", group="Boiler plant configuration parameters"));
+
+  parameter Integer nBoi = 2
+    "Number of boilers"
+    annotation(Dialog(tab="General", group="Boiler plant configuration parameters"));
+
+  parameter Integer boiTyp[nBoi]={
+    Buildings.Controls.OBC.ASHRAE.PrimarySystem.BoilerPlant.Types.BoilerTypes.condensingBoiler,
+    Buildings.Controls.OBC.ASHRAE.PrimarySystem.BoilerPlant.Types.BoilerTypes.nonCondensingBoiler}
+    "Boiler type"
+    annotation(Dialog(tab="General", group="Boiler plant configuration parameters"));
+
+  parameter Integer nSta = 3
+    "Number of boiler plant stages"
+    annotation(Dialog(tab="General", group="Boiler plant configuration parameters"));
+
+  parameter Integer staMat[nSta, nBoi] = {{1,0},{0,1},{1,1}}
+    "Staging matrix with stage as row index and boiler as column index"
+    annotation(Dialog(tab="General", group="Boiler plant configuration parameters"));
+
+  parameter Integer iniSta = 1
+    "Initial boiler plant stage"
+    annotation(Dialog(tab="General", group="Boiler plant configuration parameters"));
+
+  parameter Real boiDesCap[nBoi](
+    final unit="W",
+    final displayUnit="W",
+    final quantity="Power")
+    "Design boiler capacities vector"
+    annotation(Dialog(tab="General", group="Boiler plant configuration parameters"));
+
+  parameter Real boiFirMin[nBoi](
+    final unit="1",
+    final displayUnit="1")
+    "Boiler minimum firing ratio"
+    annotation(Dialog(tab="General", group="Boiler plant configuration parameters"));
+
+  parameter Real bMinPriPumSpeSta[nSta](
+    final unit="1",
```

Comment:
please use fill function as described here `https://obc.lbl.gov/specification/cdl.html#sec-dec-par`. For example:

```
  CDL.Interfaces.RealInput uChiWatIsoVal[nChi](
    final min=fill(0, nChi),
    final max=fill(1, nChi),
    final unit=fill("1", nChi))
    "Chilled water isolation valve position"
    annotation (Placement(transformation(extent={{-840,-220},{-800,-180}}),
      iconTransformation(extent={{-140,-50},{-100,-10}})));
```